modules/collect/src/main/java/com/opengamma/strata/collect/array/LongArray.java

```diff
@@ -0,0 +1,1297 @@
+   * @throws IndexOutOfBoundsException if the index is invalid
+   */
+  public LongArray with(int index, long newValue) {
+    if (array[index] == newValue) {
+      return this;
+    }
+    long[] result = array.clone();
+    result[index] = newValue;
+    return new LongArray(result);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Returns an instance with the specified amount added to each value.
+   * <p>
+   * This is used to add to the contents of this array, returning a new array.
+   * <p>
+   * This is a special case of {@link #map(LongUnaryOperator)}.
+   * This instance is immutable and unaffected by this method.
+   * 
+   * @param amount  the amount to add, may be negative
+   * @return a copy of this array with the amount added to each value
+   */
+  public LongArray plus(long amount) {
+    if (amount == 0) {
+      return this;
+    }
+    long[] result = new long[array.length];
+    for (int i = 0; i < array.length; i++) {
+      result[i] = array[i] + amount;
+    }
+    return new LongArray(result);
+  }
+
+  /**
+   * Returns an instance with the specified amount subtracted from each value.
+   * <p>
+   * This is used to subtract from the contents of this array, returning a new array.
+   * <p>
+   * This is a special case of {@link #map(LongUnaryOperator)}.
+   * This instance is immutable and unaffected by this method.
+   * 
+   * @param amount  the amount to subtract, may be negative
+   * @return a copy of this array with the amount subtracted from each value
+   */
+  public LongArray minus(long amount) {
+    if (amount == 0) {
+      return this;
+    }
+    long[] result = new long[array.length];
+    for (int i = 0; i < array.length; i++) {
+      result[i] = array[i] - amount;
+    }
+    return new LongArray(result);
+  }
+
+  /**
+   * Returns an instance with each value multiplied by the specified factor.
+   * <p>
+   * This is used to multiply the contents of this array, returning a new array.
+   * <p>
+   * This is a special case of {@link #map(LongUnaryOperator)}.
+   * This instance is immutable and unaffected by this method.
+   * 
+   * @param factor  the multiplicative factor
+   * @return a copy of this array with the each value multiplied by the factor
+   */
+  public LongArray multipliedBy(long factor) {
+    if (factor == 1) {
+      return this;
+    }
+    long[] result = new long[array.length];
+    for (int i = 0; i < array.length; i++) {
+      result[i] = array[i] * factor;
+    }
+    return new LongArray(result);
+  }
+
+  /**
+   * Returns an instance with each value divided by the specified divisor.
+   * <p>
+   * This is used to divide the contents of this array, returning a new array.
+   * <p>
+   * This is a special case of {@link #map(LongUnaryOperator)}.
+   * This instance is immutable and unaffected by this method.
+   *
+   * @param divisor  the value by which the array applyAsLong() divided
```

Comment:
Should be "array values are divided"

Reply:
Oops. I hand a find-and-replace operation that went rogue. I thought I'd fixed everything but missed this one.